fsw/src/generic_css_device.c

```diff
@@ -0,0 +1,281 @@
+/*******************************************************************************
+** File: generic_css_device.c
+**
+** Purpose:
+**   This file contains the source code for the GENERIC_CSS device.
+**
+*******************************************************************************/
+
+/*
+** Include Files
+*/
+#include "generic_css_device.h"
+
+static float GENERIC_CSS_TELEM_CONVERSION = 0.001; // V
+
+/*
+** Convert data received for an ADC based on predefined conversion value
+*/
+float GENERIC_CSS_TelemetryConversion(uint16 read_value)
+{
+    float rval = GENERIC_CSS_TELEM_CONVERSION * read_value;
+    return rval;
+}
+
+/* 
+** Generic read data from device
+*/
+int32_t GENERIC_CSS_ReadData(int32_t handle, uint8_t* read_data, uint8_t data_length)
+{
+    int32_t status = OS_SUCCESS;
+    uint8* response;
+
+    status = i2c_master_transaction(handle, GENERIC_CSS_I2C_ADDRESS, NULL, 0, &response, data_length, GENERIC_CSS_CFG_MS_TIMEOUT);
+    if (status != OS_SUCCESS)
+    {
+        *read_data = 0.0;
+        //return OS_ERROR;
```

Comment:
If error we shouldn't edit read_data at all, but should be reporting that an error occured.